### PR TITLE
Error response when there is problem with Mailchimp

### PR DESCRIPTION
**Pull request by andrzejewsky**
I've added error handling for Mailchimp as there is only `4**` code in the response which is hard to recognise as an error in the UI.

Also I've updated `taxtCalc` to be consisted with this one in VSF.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Force ES connections to use protocol config option - @cewald (#303, #304)
 - Better handling of HTTP error codes provided by API client - #3151
 
+### Changed
+- Error responses for mailchimp - (@andrzejewsky)
```
Comment (lukeromanowicz) [inline CHANGELOG.md]: Issue or PR id is missing, the format is `- @nickname (#issue_id)`

Comment (lukeromanowicz) [inline CHANGELOG.md]: Issue or PR id is missing, the format is `- @nickname (#issue_id)`

File: src/lib/taxcalc.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ function isSpecialPriceActive (fromDate, toDate) {
   }
 }
 
-export function updateProductPrices (product, rate, sourcePriceInclTax = false, deprecatedPriceFieldsSupport = false, finalPriceInclTax = true) {
+export function updateProductPrices ({ product, rate, sourcePriceInclTax = false, deprecatedPriceFieldsSupport = false, finalPriceInclTax = true }) {
```
Comment (lukeromanowicz) [inline src/lib/taxcalc.js]: @andrzejewsky please add a note about that breaking change in the changelog. There might be some third-party platforms that also are using this method.

**Comment by andrzejewsky** [comment]
@lukeromanowicz done.

**Pull request merged**